components/form.js

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import Link from 'next/link';
+import { Person } from './Users';
+
+const baseUrl = 'https://rominacsvsms.api.ushahidi.io';
+
+export default class Form extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            token: {
+                access_token: '',
+                expires_in: 0,
+                refresh_token: '',
+                token_type: ''
+            },
+            formFields: [],
+            formData: {}
+        };
+    }
+
+    // Make state targets equal to the value of input fields
+    change = e => {
+        this.setState({
+            formData: {
+                ...this.state.formData,
+                [e.target.name]: e.target.value
+            }
+        });
+    };
+
+    isTokenValid() {
+        let isValid = false;
+        const currentTime = new Date().getTime();
+        const tokenFetchTime = this.state.token.fetching_time;
+        const expirationTime = this.state.token.expires_in;
+        if (currentTime < tokenFetchTime + expirationTime) {
+            isValid = true;
+        }
+        return isValid;
+    }
+
+    getToken() {
+        let token = JSON.parse(localStorage.getItem('Bearer Token'));
+        if (!token || this.isTokenValid()) {
+            fetch(baseUrl + '/oauth/token', {
+                method: 'POST',
+                body: JSON.stringify({
+                    scope: '*',
+                    client_secret: '35e7f0bca957836d05ca0492211b0ac707671261',
+                    client_id: 'ushahidiui',
+                    grant_type: 'password',
+                    password: Person.password,
+                    username: Person.userName
+                }),
+                headers: {
+                    Accept: 'application/json',
+                    'Content-Type': 'application/json',
+                    Authorization: 'large-auth-string'
```

Comment:
please don't use placeholder text. Use `null` to signify a null value (here and elsewhere)

Reply:
also for this request in particular you don't need to sent the auth header since you are requesting a token , sorry I didn't mention that !